lib.go

```diff
@@ -192,9 +191,12 @@ func getFieldByTag(ival interface{}, tagName string, fieldNames []string) (inter
 
 		parts := strings.Split(tagValue, ",")
 		if parts[0] == fieldName {
```

Comment:
`strings.Cut` is in 1.18 and avoids allocating a whole slice

```suggestion
		if tagFieldName, _, _ = strings.Cut(tagValue); tagFieldName == fieldName {
```